framework/OutStreams/PrintInterfaces/PrintInterface.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 2, 2021
+
+@author: talbpaul
+"""
+
+from OutStreams import OutStreamInterface
+
+class PrintInterface(OutStreamInterface):
+  """
+    Archetype for Print implementations
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'PrintInterface'
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+
+  def initialize(self, stepEntities):
+    """
+      Function to initialize the OutStream.
+      @ In, stepEntities, dict, contains all the Objects are going to be used in the
+                                current step. The sources are searched into this.
+      @ Out, None
+    """
+    super().initialize(stepEntities)
+
+  def run(self):
```

Comment:
This method is not need here, it is defined the BaseInterface class, right?

Reply:
I believe it is wise to include this for clarity, given the developers will inherit from this interface.